source/PlayerInfo.cpp

```diff
@@ -577,6 +577,11 @@ void PlayerInfo::IncrementDate()
 	string message = accounts.Step(assets, Salaries());
 	if(!message.empty())
 		Messages::Add(message);
+	
+	// Advance the reload counters on all the player's weapons.
+	int advanceFrames = 24 * 60 * 60 * 60;	
```

Comment:
I think this (and maybe some others) should be const int or even static const int? @endless-sky will know more.

Looks good re functionality - got a simple test case we can use to verify the issue (and the fix?)